flink-core/src/main/java/org/apache/flink/api/common/io/FileInputFormat.java

```diff
@@ -235,7 +236,12 @@ public FileInputFormat() {}
 
     protected FileInputFormat(Path filePath) {
         if (filePath != null) {
-            setFilePath(filePath);
+            // support glob
+            setFilePath(filePath.getParent());
+            setFilesFilter(
+                    new GlobFilePathFilter(
+                            Collections.singletonList(filePath.getPath()),
+                            Collections.emptyList()));
```

Comment:
Does it make sense to check if there is a `*` and only use the glob code then? Note sure if there is a noticeable performance penalty (I guess not since that is probably only setup code).